### PR TITLE
Add diff_drive_controller and wheel odometry (integrated with Kalman Filter)

### PR DESCRIPTION
**Pull request by blicogam**
In order to add wheel odometry, the controllers to move to rover were changed. The wheel odometry has also been integrated into the Kalman Filter.

- Instead of a JointVelocityController for each wheel, a single diff_drive_controller is used to control all four wheels.
- This controller expects a single Twist command for the whole rover instead of a Float64 for each wheel. This makes it more user friendly as Twist is the common way to move a robot. The original implementation was essentially doing the diff_drive_controller math to state how much each wheel has to turn and what direction.
- Wheel odometry is automatically calculated for us. No need to "simulate" encoders to find out how much each wheel has turned and calculate the odometry from that.

I think diff_drive_controller is built into ros but otherwise:
`sh develop.sh resolve`

And then run the simulation:
`roslaunch ezrassor_launcher configurable_simulation.launch control_methods:="autonomy" enable_real_odometry:="true"`

Then, in another terminal, to auto drive to dig site ( default is (10, 10) ):
`rostopic pub /ezrassor1/autonomous_toggles std_msgs/Int8 1`

To see the actual output of the Kalman filter, open another terminal and type:
`rostopic echo /ezrassor1/odometry/filtered`

You can change the spawn coordinates and dig site coordinates by adding the following flags in the simulation launch command:
`spawn_x_coords:=some_new_x spawn_y_coords:=some_new_y`
`digsite_x_coords:=some_new_x digsite_y_coords:=some_new_y`

NOTE: The dig site coordinates are in the rover's local frame. So if the rover spawns in any place other than the origin, it will not go to (10, 10), but rather, go to the position that is 10 in front of you and 10 to the left of you. In a future pull request, this will be changed to the global frame but it's important to note if dealing with something such as path planning.

There are optional constraints for linear and angular in the diff_drive_controller but only linear is used in this case. The linear constraints listed in code don't line up exactly with the mini-RASSOR (1m/s as opposed to the real 1m/3s) but are suited to be more responsive for faster testing in simulation. 

Prior to actual implementation on physical hardware, developers should adjust these optional constraints in simulation to ensure they suit their application.

### CODE REVIEW

File: packages/autonomy/ezrassor_autonomous_control/launch/ekf_template.yaml [hunk 1]
```diff
@@ -19,3 +19,14 @@ imu0_relative: true
 imu0_queue_size: 5
 
 imu0_remove_gravitational_acceleration: true
+
```
Comment (tgsachse) [inline packages/autonomy/ezrassor_autonomous_control/launch/ekf_template.yaml]: Can you write launch files in yaml? Or is this more of a configuration file? If it's configuration, is it possible for this to go in a config directory?

File: packages/simulation/ezrassor_sim_description/urdf/ezrassor.gazebo [hunk 2]
```diff
@@ -122,6 +122,9 @@
 	  </plugin>
   </gazebo>
 
+
```
Comment (tgsachse) [inline packages/simulation/ezrassor_sim_description/urdf/ezrassor.gazebo]: nit: what's this extra whitespace for?

**Comment by jalbury** [comment]
@tgsachse Just made those changes you talked about. The YAML file is a configuration file for the Kalman filter node, so it makes sense for it to be in a separate config directory.

**Comment by tgsachse** [comment]
You need to add the config directory to the CMakeList installation file or it won't be available when installed and will break. Here's an example from one of the other packages: https://github.com/FlaSpaceInst/EZ-RASSOR/blob/master/packages/communication/ezrassor_joy_translator/CMakeLists.txt

Make that change and test the install procedure, then approved.

**Comment by jalbury** [comment]
@tgsachse Ah okay, didn't realize that would break the installation. Just made those changes and confirmed the install procedure still works. Thanks for your help!

**Pull request merged**